xbmc/cores/VideoPlayer/DVDCodecs/Overlay/OverlayCodecWebVTT.cpp

```diff
@@ -124,7 +135,7 @@ void COverlayCodecWebVTT::Reset()
 
 void COverlayCodecWebVTT::Flush()
 {
-  if (m_isISOFormat)
+  if (m_allowFlush)
```

Comment:
Why is this not `m_isISOFormat || m_allowFlush`

Reply:
because i have inverted the flush execution, so now by default is enabled, and will be blocked only on "file" mode

Reply:
I thought that if in the future ffmpeg will rework  his webvtt demux implementation
its packages should require the flush enabled by default

Reply:
Yeah I think this is intended behaviour, we want to allow flush in certain cases (e.g. file not in extradata) and not depending on the format.

Reply:
Where is the inversion. I think I’m missing something.

Reply:
before was `m_isISOFormat` by default set as false
now use `m_allowFlush` by default set as true

Reply:
no wait...I'm getting confused

Reply:
Yeah, if that was intended I think it's correct.
However:
>I thought that if in the future ffmpeg will rework his webvtt demux implementation
its packages should require the flush enabled by default

Not sure if I understand the sentence correctly:
We shouldn't care what ffmpeg might do in the future, we can change behavior once there
is need caused by ffmepg reworks - not prematurely because it might be needed in the future.
How do you know the new behavior doesn't break other stuff?

Reply:
In that case you also flush now even if there is no extra data which changes behaviour

Reply:
recap because can be confusing,
some other is changed on ISAdaptive that affect this,
previous situation:
flush was required to:
- ISO format

and was not required for:
- Webvtt segmented
- Webvtt as file

Now on ISA Webvtt segmented require the flush (sending packets bug fixed)
then only the "file" mode still require to avoid the flush

The new situation then is this,
flush required to:
- ISO format
- Webvtt segmented

and not required for:
- Webvtt as file

then to me seem right to have flush enabled by default, and disabled only on "file" mode

Reply:
I can follow that logic, so this looks fine to me.
But how about non ISA usecases - is `COverlayCodecWebVTT` used in more generic cases as well, which might break because of this?

Reply:
I know only that this class could be used by ffmpeg subtitle demuxer, but only when in ffmpeg will be reworked the current outdated webvtt implementation (apparently not a priority on ffmpeg and no one knows when it will be done),
this use case is indeed to allow use webvtt with containers like MKV files

there is also webvtt ISO support for MP4 files (fmp4 data), maybe i am wrong but to demux this case, seems to require the support from libraries that we don't have in kodi

Reply:
Apparently you've researched in great depth and you can argue your implementation, definitely you know way more about it than I do, so from my pov it's fine.
Also we're in (pre-)alpha, so if it breaks we can fix it.